admin-dev/themes/new-theme/js/pages/product/edit/product-type-switcher.ts

```diff
@@ -59,6 +59,10 @@ export default class ProductTypeSwitcher {
     this.initialType = <string> this.$typeSelector.val();
 
     this.$productTypePreview.on('click', () => this.showSelectionModal());
+    $('input').on('change', () => {
```

Comment:
With this it means as soon as something is changed on the page you can't use the button ever again. Even if you revert the modifications. We have a component that checks the product state more accurately here: https://github.com/PrestaShop/PrestaShop/blob/057111abc17f0e0088409f1f829011a2ae43fd0f/admin-dev/themes/new-theme/js/pages/product/edit/product-partial-updater.ts#L226

It's where we toggle the footer buttons' states, so that's where you should toggle the switch component as well. You'll need to pass the `ProductTypeSwitcher` as an argument to the updater, and add a `toggle` method or two enable/disable methods to drive it.

Reply:
while trying to use this logic noticed that in getUpdatedFormData there is always some info that is seen as changed because its null and is added empty string.
![image](https://user-images.githubusercontent.com/51944825/194544946-c7b48914-9e57-42f6-83c4-60b4648acff8.png)

in the image you can see what values are seen as changed. Investigating if it works fine if we dont change null to empty string. Not sure if this should be new issue or can be added on this one.